### DISCUSSION TITLE
Apple pay with own certificate

### DISCUSSION THREAD

**Discussion by eloy98g** [Adyen/adyen-react-native#120] [Q&A] [answered]
Hi, is there any configuration needed for showing apple pay with our own certicicate on iOS?

We are using the "DropIn" Component, and the payment_methods we have for iOS devices are card and apple pay, but only card is showing. Apple pay is set up with our own certificate in our dashboard and the only configuration we have for it inside the app code is an object:
`applepay: { merchantIdentifier: MERCHANTID, merchantName: MERCHANTNAME}`

We have adyen-web aswell (as we have web and mobile platforms in the same project) and we had to implement the onValidateMetchant logic on applepay configuration. Do we need something like that for mobile platforms? i couldn't find any docs for this use-case. 

When we are using emulators it does show up.

Thanks!

**Comment by descorp** [comment, marked as answer]
Hey @eloy98g 

Thanks for starting this discussion.

You can find documentation [here](https://docs.adyen.com/payment-methods/apple-pay/enable-apple-pay?tab=i_os_2), but it is mostly referring to Apple's documentation.

> `onValidateMetchant` logic on applepay configuration

Merchant validation is not needed on iOS, since your AppID [contains merchant identifier](https://developer.apple.com/help/account/configure-app-capabilities/configure-apple-pay#create-a-merchant-identifier), hence Xcode will validate it during build/app signing.

> we have for iOS devices are card and apple pay, but only card is showing

For ApplePay to work, you also need to provide `countryCode` and `amount` on [configuration](https://github.com/Adyen/adyen-react-native#configuration).

**Comment by eloy98g** [comment]
Thanks! problem solved adding merchantIDs with Xcode.

**Comment by todibbang** [comment]
Hi @descorp!
I have a related issue.
I'm trying to get Apple Pay to show up in checkout, but I'm unable to get it to work.

I have an Adyen platform with multiple merchants attached. 
I'm using our own apple certificate.
I'm currently using "@adyen/react-native": "1.2.0"

This is what I've done:
1. Created a merchant id in https://developer.apple.com/ and added it as a capability in xcode:
<img width="756" alt="Screenshot 2023-12-21 at 14 08 53" src="https://github.com/Adyen/adyen-react-native/assets/12685946/a35b357f-1400-4ab7-8f9e-e96b8d12d580">

2. Added apple pay as a payment method in Adyen/Settings/Payment methods for the specific merchant that I'm trying to enable apple pay for:
<img width="803" alt="Screenshot 2023-12-21 at 14 00 13" src="https://github.com/Adyen/adyen-react-native/assets/12685946/c26f8e53-5d5e-4649-9099-f3e98581b9f9">

3. Configured the apple pay certificate in the API credential settings:
<img width="1207" alt="Screenshot 2023-12-21 at 14 04 03" src="https://github.com/Adyen/adyen-react-native/assets/12685946/73d21ca7-939a-423f-9786-670fe8ad6186">

4. Verified that country code and amount are added to the adyen checkout config:
`countryCode: DK, paymentAmount {"value":20000,"currency":"DKK"}`
<img width="334" alt="Screenshot 2023-12-21 at 14 04 42" src="https://github.com/Adyen/adyen-react-native/assets/12685946/33733803-78ce-4ed0-81ac-e22099321315">

6. Verified that I do receive apple-pay as a payment method when initialising a payment, with matching client-id and merchant-id:
<img width="522" alt="Screenshot 2023-12-21 at 14 08 00" src="https://github.com/Adyen/adyen-react-native/assets/12685946/633526e3-4e0f-4993-9810-7e9d60e39e0c">

7. Tested it both locally and as a deployed build from testflight.
<img width="402" alt="Screenshot 2023-12-21 at 14 14 01" src="https://github.com/Adyen/adyen-react-native/assets/12685946/c5007a68-45df-47c3-840a-c84be995ba46">


I have **not** done the step described here, as this appear to be for using apple pay using a website?
https://docs.adyen.com/payment-methods/apple-pay/enable-apple-pay/?tab=i_os_2
<img width="832" alt="Screenshot 2023-12-21 at 14 11 20" src="https://github.com/Adyen/adyen-react-native/assets/12685946/f16639ec-4ca2-4545-aa02-603111c56adb">

I have noticed this warning when initialising the Adyen payment, not sure if it's related?
`Task <8610EC81-0DCA-48D5-882C-48DBA2C78C1D>.<18> finished with error [18.446.744.073.709.550.613] Error Domain=NSURLErrorDomain Code=-1003 "A server with the specified hostname could not be found." UserInfo={_kCFStreamErrorCodeKey=8, NSUnderlyingError=0x2837e66a0 {Error Domain=kCFErrorDomainCFNetwork Code=-1003 "(null)" UserInfo={_kCFStreamErrorDomainKey=12, _kCFStreamErrorCodeKey=8, _NSURLErrorNWResolutionReportKey=Resolved 0 endpoints in 0ms using unknown from cache, _NSURLErrorNWPathKey=satisfied (Path is satisfied), interface: en0[802.11], ipv4, dns, uses wifi}}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <8610EC81-0DCA-48D5-882C-48DBA2C78C1D>.<18>, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <8610EC81-0DCA-48D5-882C-48DBA2C78C1D>.<18>"
), NSLocalizedDescription=A server with the specified hostname could not be found., NSErrorFailingURLStringKey=https://checkoutshopper-live-eu.adyen.com/checkoutshopper/v2/analytics/id?clientKey=live_12....34, NSErrorFailingURLKey=https://checkoutshopper-live-eu.adyen.com/checkoutshopper/v2/analytics/id?clientKey=live_12....34, _kCFStreamErrorDomainKey=12}`

Any idea where things might have gone wrong? :)

**Comment by descorp** [comment]
Hey @todibbang 

You need to specify `merchantID` and/or `merchantName` in **config**:

```js
<AdyenCheckout
  config={{
    countryCode: 'NL',
    amount: {
      currency: 'EUR',
      value: 9800,
    },
    applepay: {
      merchantID: '{YOUR_APPLE_MERCHANT_ID}',
      merchantName: '{YOUR_APP_NAME}',
    },
    ...
  }},
  ...
/>;

```

See full Applepay configuration [here](https://github.com/Adyen/adyen-react-native/blob/develop/docs/Configuration.md#example)

**Reply by todibbang** [reply]
Okay, now the original comment makes sense to me too. Thank you for clarifying, it works!
